topo/topo.go

```diff
@@ -256,6 +258,55 @@ func (m *Manager) Push(ctx context.Context) error {
 	return nil
 }
 
+// CheckNodeStatus reports node status, ignores for unimplemented nodes.
+func (m *Manager) CheckNodeStatus(ctx context.Context) error {
+	foundAll := false
+	failed := false
+	nodes := make(map[string]*node.Node)
+	for k, n := range m.nodes {
+		nodes[k] = n
+	}
+
+	// Check until end state or 3 sec expired
+	for i := 0; i < 30 && !foundAll; i++ {
```

Comment:
Why are we saying wait for 3 secs? feels like we should either block until a state or have a flag provided timeout to wait?

what is our guarantee that this everything is in the right status?

Reply:
Will provide a timeout <seconds> flag.

For successful status the Reason would be empty. Will leave it up to node specific implementation, to interpret overall successful status.

Reply:
Just to clarify, for Ixia we will verify Pod and Service status before we do a status update Success.